### PR TITLE
Adds Webpack support and remove Grunt

### PR DESCRIPTION
**Pull request by tcitworld**
So I was rather pissed off by Grunt being so slow and I didn't want to look into livereload plugins, so I went directly for the webpack hype.
So basically, this brings up latest Materialize (ew) and all up to date versions of libraries (except good old jQuery which is required to be 2.x by Materialize).

And you get a `yarn.lock` file and `webpack -w` when working on front-end.

- [x] I should take advantage from this PR to fix a few things in the baggy theme (poor old baggy)
- [x] Translate datepicker => separate issue
- [x] Bring babel for old browsers
- [x] Clean up old css code : Move to SASS ?
- [x] Check if we could remove IcoMoon.
- [x] Fix share buttons in material
- [x] Add webpack ExtractTextPlugin for production
- [x] Add webpack-dev-server for assets live-reload !
- [x] Add manifest.json (needs sf 3.3) http://symfony.com/blog/new-in-symfony-3-3-manifest-based-asset-versioning
- [x] Fix new article / search bar in material theme

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | should
| Documentation | no
| Translation   | no
| Fixed tickets | 
| License       | MIT

### CODE REVIEW

File: web/bundles/wallabagcore/manifest.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  "img/carrot-icon--white.png": "img/carrot-icon--white.png",
+  "img/diaspora-icon--black.png": "img/diaspora-icon--black.png",
+  "img/list.png": "img/list.png",
+  "img/shaarli.png": "img/shaarli.png",
+  "img/table.png": "img/table.png",
+  "img/unmark-icon--black.png": "img/unmark-icon--black.png",
+  "material.css": "material.css",
+  "material.css.map": "material.css.map",
+  "material.js": "material.js"
+}
```
Comment (j0k3r) [inline web/bundles/wallabagcore/manifest.json]: New line needed.
Or maybe it's a generated file and then we don't care.

File: app/config/config.yml [hunk 2]
```diff
@@ -3,6 +3,9 @@ imports:
     - { resource: security.yml }
     - { resource: services.yml }
 
+parameters:
+    use_webpack_dev_server: false
```
Comment (j0k3r) [inline app/config/config.yml]: Maybe add an extra comment here to explain what it activate

File: package.json [hunk 3]
```diff
@@ -1,34 +1,13 @@
 {
   "name": "wallabag",
-  "version": "2.1.0",
+  "version": "2.2.2",
```
Comment (j0k3r) [inline package.json]: Do we really need to put a version number here?

File: web/bundles/wallabagcore/manifest.json (hunk 1 above)
Comment (tcitworld) [inline web/bundles/wallabagcore/manifest.json]: Yup, generated.

File: package.json (hunk 3 above)
Comment (tcitworld) [inline package.json]: The `version` field is required, but it doesn't really matter what we put in it.

**Comment by j0k3r** [comment]
Looks like I've some issue with select in material design:

**Comment by tcitworld** [comment]
Fixed the theme issue.

**Comment by tcitworld** [comment]
Needs travis changes, hold on...

File: .travis.yml [hunk 4]
```diff
@@ -75,4 +74,4 @@ script:
     - if [[ $VALIDATE_TRANSLATION_FILE = run ]]; then php bin/console lint:yaml src/Wallabag/CoreBundle/Resources/translations -v ; fi;
     - if [[ $VALIDATE_TRANSLATION_FILE = run ]]; then php bin/console lint:yaml app/Resources/CraueConfigBundle/translations -v ; fi;
     - if [[ $VALIDATE_TRANSLATION_FILE = run ]]; then php bin/console lint:yaml src/Wallabag/UserBundle/Resources/translations -v ; fi;
-    - if [[ $ASSETS = build ]]; then ./node_modules/grunt-cli/bin/grunt tests; fi;
+    - if [[ $ASSETS = build ]]; then ./node_modules/webpack/bin/webpack.js; fi;
```
Comment (j0k3r) [inline .travis.yml]: Instead of webpack, the build shouldn't run `yarn run build:dev` ?

Comment (tcitworld) [inline .travis.yml]: More like `yarn run build:prod`, but yes.

Comment (tcitworld) [inline .travis.yml]: Fixed

**Comment by nicosomb** [comment]
This PR must target `2.3`.

**Comment by tcitworld** [comment]
Oh, -20% file size from the old js files. Nice.

**Comment by tcitworld** [comment]
Can we merge this before I have a look at https://github.com/wallabag/wallabag/pull/3077 ?

**Comment by j0k3r** [comment]
Hum

image

Looks like link on tag are white :confused: 

image

**Comment by j0k3r** [comment]
Looks like icon in tag list of an entry are a bit different.

before:
image

after:
image

**Comment by j0k3r** [comment]
Also, it looks like the margin bug in the menu when hovering an item (both on the dashboard and when viewing an entry) is back:

before
image

after
image

**Comment by tcitworld** [comment]
Should be fixed now.

**Comment by j0k3r** [comment]
Some tabs are going _under the hood_ in the config panel & internal settings:

image

My resolution is 2560 x 1600 Retina (so I guess divide by 2 🤔)

**Comment by tcitworld** [comment]
Sorry, looked with all kind of resolutions, but can't reproduce.

**Comment by j0k3r** [comment]
Just tested on bigger screen and got the same result.
Did you see the `Tagging rules` in config?

And I don't think it's a resolution problem:
image

**Comment by nicosomb** [comment]
#### About config, I don't have Password tab. 

capture d ecran 2017-05-05 a 15 51 02

The Tagging rules one appears when I unzoomed. 

#### RSS part needs some spaces: 

capture d ecran 2017-05-05 a 15 49 48
capture d ecran 2017-05-05 a 15 49 58

#### 😍

Finally!

capture d ecran 2017-05-05 a 15 52 49

#### Share menu on entry view is broken 

capture d ecran 2017-05-05 a 15 53 21


#### On list view, icons part is broken

capture d ecran 2017-05-05 a 15 54 34

#### Hover on logo: I think we should remove this effect 

capture d ecran 2017-05-05 a 15 55 10

**Comment by nicosomb** [comment]
My review seems to be fixed 👍 good job "Aline" 😁

**Comment by nicosomb** [comment]
On article view, the icon for "original url" has "..." : 

capture d ecran 2017-05-07 a 15 23 14

Tags need some space

capture d ecran 2017-05-07 a 15 24 00

**Comment by nicosomb** [comment]
capture d ecran 2017-05-07 a 17 43 17

The 3dots menu seems broken, right?

**Comment by tcitworld** [comment]
> The 3dots menu seems broken, right?

Works perfectly for me. :/

**Comment by tcitworld** [comment]
Fixed.

**Comment by j0k3r** [comment]
image

Email icon seems bigger than before and also with a little margin on the left.
Otherwise, it's ok :slightly_smiling_face:

**Pull request merged**